### PR TITLE
Convert OpenMPTarget ParallelReduce and ParallelScan

### PR DESCRIPTION
**Pull request by masterleinad**
Part of #5908.

### CODE REVIEW

File: core/src/OpenMPTarget/Kokkos_OpenMPTarget_ParallelScan_Range.hpp [hunk 1]
```diff
@@ -76,10 +77,12 @@ class ParallelScan<FunctorType, Kokkos::RangePolicy<Traits...>,
     idx_type nteams           = n_chunks > 512 ? 512 : n_chunks;
     idx_type team_size        = 128;
 
-    FunctorType a_functor(m_functor);
-#pragma omp target teams distribute map(to : a_functor) num_teams(nteams)
+    auto a_functor_reducer = m_functor_reducer;
+#pragma omp target teams distribute map(to \
+                                        : a_functor_reducer) num_teams(nteams)
```
Comment (rgayatri23) [inline core/src/OpenMPTarget/Kokkos_OpenMPTarget_ParallelScan_Range.hpp:82]: We should be able to get rid of this indirection of `a_functor_reducer` and directly pass `m_functor_reducer` . But maybe that can be done separately.

**Review by rgayatri23 — APPROVED** [review]
Looks like the changes are only to the function signatures and the associated calls within routines to make it consistent with the new reduction interface. 

LGTM.

**Comment by Rombur** [comment]
You need to rebase this PR. There is a conflict.

**Review by Rombur — APPROVED** [review]

**Comment by dalg24** [comment]
Ignoring SYCL failure.  HIP builds did not run.

**Pull request merged**